src/mapml.js

```diff
@@ -916,15 +1018,17 @@ M.MapMLLayer = L.Layer.extend({
         // the layer._imageContainer property contains an element in which
         // content will be maintained
         
-        if (!this._tileLayer) {
-          this._tileLayer = M.mapMLTileLayer(this.href?this.href:this._href, 
-          {pane: this._container,
-           _leafletLayer: this});
-        }
-        // what is _mapmlTileContainer for and why have two copies of it.
-        this._tileLayer._mapmlTileContainer = this._mapmlTileContainer;
-        map.addLayer(this._tileLayer);       
-        this._tileLayer._container.appendChild(this._mapmlTileContainer);
+        //only add the layer if there are tiles to be rendered
+        if((!this._staticTileLayer || this._staticTileLayer._container === null) && this._mapmlTileContainer.getElementsByTagName("tiles").length > 0){
```

Comment:
Style comment: can we break up really long lines of code onto multiple lines, with indenting so as to show that the next line belongs to the previous if appropriate.  Break after operators and at logical points so as to try to preserve readability.  Thanks!